src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -171,6 +171,8 @@ TxnNamedRead createNamedRead(NamedSelect namedSelect, QueryOptions options)
     {
         SelectStatement select = namedSelect.select;
         ReadQuery readQuery = select.getQuery(options, 0);
+        if (!(readQuery instanceof SinglePartitionReadQuery.Group))
+            throw new IllegalArgumentException("Partition range queries are illegal in transactions");
```

Comment:
nit: I'd define a constant message...

```
public static final String ILLEGAL_RANGE_QUERY_MESSAGE = "Range queries are not allowed for reads within a transaction";
```
...and then replace all three of these w/ something like...

```
checkTrue(readQuery instanceof SinglePartitionReadQuery.Group, ILLEGAL_RANGE_QUERY_MESSAGE)
```